flight/UAVObjects/uavobjectmanager.c

```diff
@@ -1658,24 +1681,58 @@ void UAVObjIterate(void (*iterator) (UAVObjHandle obj))
 	PIOS_Recursive_Mutex_Unlock(mutex);
 }
 
-/**
- * Send a triggered event to all event queues registered on the object.
- */
-static int32_t sendEvent(struct UAVOBase * obj, uint16_t instId,
-			UAVObjEventType triggered_event)
-{
-	/* Set up the message that will be sent to all registered listeners */
-	UAVObjEvent msg = {
-		.obj    = (UAVObjHandle) obj,
-		.event  = triggered_event,
-		.instId = instId,
-	};
+/* type signature must match invokeCallback below, with 4 or fewer args */
+static void __attribute__((used)) realInvokeCallback(struct ObjectEventEntry *event,
+		UAVObjEvent *msg, void *obj_data, int len);
 
+static void realInvokeCallback(struct ObjectEventEntry *event,
+		UAVObjEvent *msg, void *obj_data, int len) {
+	event->cb(msg, event->cbInfo.cbCtx, obj_data, len);
+}
+
+#if (!defined(SIM_POSIX)) && defined(__arm__)
+static void invokeCallback(struct ObjectEventEntry *event, UAVObjEvent *msg,
```

Comment:
Can I ask that you comment the crap out of this? I admire the assembly code and appreciate its need, but I worry that it's not easily maintainable.

I know you've already done a lot, but even with the comments I'm not sure of 1) my ability to review the code for sanity and 2) my ability to maintain the code if necessary.